### PR TITLE
Select multiples of a mod in mod picker

### PR DESCRIPTION
**Pull request by ryan-rushton**
Allow the selection of more than one armour 2 mod in the new mod picker.

image
image

### CODE REVIEW

File: src/app/loadout-builder/ModPickerFooter.m.scss [hunk 1]
```diff
@@ -4,7 +4,7 @@
   composes: flexRow from '../dim-ui/common.m.scss';
   align-items: center;
 
-  > * {
+  div {
```
Comment (bhollis) [inline src/app/loadout-builder/ModPickerFooter.m.scss:7]: Without the descendent selector this'll match too much

File: src/app/loadout-builder/types.ts [hunk 2]
```diff
@@ -72,6 +72,7 @@ export const ModPickerCategories = { ...Armor2ModPlugCategories, seasonal: 'seas
 export type ModPickerCategory = typeof ModPickerCategories[keyof typeof ModPickerCategories];
 
 export interface LockedArmor2Mod {
+  key: string;
```
Comment (bhollis) [inline src/app/loadout-builder/types.ts:75]: It'd be nice to document what this is.

File: src/app/loadout-builder/ModPicker.tsx [hunk 3]
```diff
@@ -198,7 +199,7 @@ class ModPicker extends React.Component<Props, State> {
             ? specialtyModSocketHashes.includes(mod.plug.plugCategoryHash)
             : mod.plug.plugCategoryHash === category
         )
-        .map((mod) => ({ mod, category }));
+        .map((mod) => ({ key: uuidv4(), mod, category }));
```
Comment (bhollis) [inline src/app/loadout-builder/ModPicker.tsx:202]: You don't really need a whole UUID - just a global counter.

Comment (ryan-rushton) [reply src/app/loadout-builder/ModPicker.tsx]: Done

File: src/app/loadout-builder/types.ts (hunk 2 above)
Comment (ryan-rushton) [reply src/app/loadout-builder/types.ts]: Done, it's basically an id of sorts for removing items and using as a jsx key. Put a comment on it, let me know if you want to rename it too. I didn't want to use id and couldn't come up with anything I liked better.

File: src/app/loadout-builder/ModPickerFooter.m.scss (hunk 1 above)
Comment (ryan-rushton) [reply src/app/loadout-builder/ModPickerFooter.m.scss]: Done, I pushed it down to where I needed it so there is some separation between mods.

**Pull request merged**